runtime-decompiler/src/main/java/org/jrd/frontend/frame/settings/SettingsView.java

```diff
@@ -67,65 +80,132 @@
-            gbc.gridy = 1;
-            this.add(Box.createHorizontalStrut(20), gbc);
-
+            gbc.insets = new Insets(5,20,0,0);
             gbc.weightx = 1;
             gbc.gridx = 1;
             this.add(agentPathTextField, gbc);
 
-            gbc.weightx = 0;
+            gbc.insets = new Insets(0,20,0,20);
             gbc.gridx = 2;
-            this.add(Box.createHorizontalStrut(20), gbc);
-
-            gbc.gridx = 3;
+            gbc.gridy = 1;
             this.add(browseButton, gbc);
 
-            gbc.gridx = 4;
-            this.add(Box.createHorizontalStrut(20), gbc);
-
-            gbc.gridx = 0;
-            gbc.gridy = 2;
-            this.add(Box.createVerticalStrut(20), gbc);
-
+            gbc.insets = new Insets(20,20,0,0);
             gbc.gridx = 1;
             gbc.gridy = 3;
             this.add(checkBoxSettings, gbc);
 
-            gbc.gridx = 0;
-            gbc.gridy = 4;
-            this.add(Box.createHorizontalStrut(20), gbc);
-
+            gbc.insets = new Insets(5,20,0,0);
             gbc.gridx = 1;
+            gbc.gridy = 4;
             this.add(useHostSystemClassesCheckBox, gbc);
 
-            this.setPreferredSize(new Dimension(0, 150));
+            // Nested Jars
+            nestedJars = new JLabel("Nested Jars Settings:");
+            newExtensionsTextField = new JTextField();
+
+            dList = new DefaultListModel();
+            list = new JList(dList);
+            list.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+            list.setLayoutOrientation(JList.VERTICAL);
+            list.setVisibleRowCount(-1);
+
+            scrollPane = new JScrollPane(list);
+            scrollPane.setPreferredSize(new Dimension(0, 200));
+
+            addButton = new JButton("ADD");
+            addButton.addActionListener(actionEvent -> {
+                for (String s : newExtensionsTextField.getText().split( "\\s")) {
+                    dList.addElement(s);
+                }
+                newExtensionsTextField.setText("");
+            });
+
+            removeButton = new JButton("REMOVE");
+            removeButton.addActionListener(actionEvent -> {
+                int index = list.getSelectedIndex();
+                dList.removeElementAt(index);
+            });
+
+            useDefaults = new JCheckBox("Use default extensions");
+            useDefaults.addItemListener(itemEvent -> {
+                newExtensionsTextField.setEnabled(!useDefaults.isSelected());
+                addButton.setEnabled(!useDefaults.isSelected());
+                removeButton.setEnabled(!useDefaults.isSelected());
+                list.setEnabled(!useDefaults.isSelected());
+            });
+            useDefaults.setToolTipText(BytecodeDecompilerView.styleTooltip() + "Default extensions that are searched are: .zip, .jar, .war, .ear");
+
+            // Setup
+            List<String> l = ArchiveManagerOptions.getInstance().getExtensions();
+            if (l == null || l.isEmpty()) {
```

Comment:
You are using one less condition for determining the usage of defaults than in `ArchiveManagerOptions#isInner()`. For this confusion alone, I would recommend to extract that check into a method in `ArchiveManagerOptions` and to use that.